services/QuillLMS/client/app/bundles/Teacher/components/progress_reports/diagnostic_reports/diagnostics/shared.tsx

```diff
@@ -24,11 +24,42 @@ export function goToAssign(unitTemplateId, name, activityId) {
   window.location.href = `/assign/select-classes?diagnostic_unit_template_id=${unitTemplateIdString}`
 }
 
+export function sumProficiencyScores(proficiencyScoresByStudent: { [name: string]: { pre: number, post: number } }, type: string) {
+  return Object.values(proficiencyScoresByStudent).reduce((total, student_score) => total += student_score[type], 0)
+}
+
+export function calculateClassGrowthPercentage({ skillGroupSummaries, completedStudentCount, setClasswideGrowthAverage }) {
+  let preTestTotal = 0
+  let postTestTotal = 0
+  const summariesCount = skillGroupSummaries.length
+  skillGroupSummaries.forEach(summary => {
+    const { proficiency_scores_by_student } = summary
+    const preScoresSum = sumProficiencyScores(proficiency_scores_by_student, PRE)
+    const postScoresSum = sumProficiencyScores(proficiency_scores_by_student, POST)
+    // we don't want to account for no growth instances so we add the pre test average for both totals
+    if(preScoresSum > postScoresSum) {
+      const sum = preScoresSum / completedStudentCount
+      preTestTotal += sum
+      postTestTotal += sum
+    } else {
+      preTestTotal += (preScoresSum / completedStudentCount)
+      postTestTotal += (postScoresSum / completedStudentCount)
+    }
+  })
+  preTestTotal = preTestTotal / summariesCount
+  postTestTotal = postTestTotal / summariesCount
+  const classAverage = Math.round((postTestTotal - preTestTotal) * 100)
+  setClasswideGrowthAverage(classAverage)
+}
```

Comment:
It took me a while to figure out what this code is doing.  I don't know that I think it needs to be refactored, but I think I'd appreciate a comment near the top of the function explaining roughly what the expected process is here so that if this ever needs to be changed someone knows what the intent is.

Reply:
that's understandable-- I've add an example payload and a bit more explanation in a comment